source/views/streaming/convocations/archived.js

```diff
@@ -0,0 +1,244 @@
+
+    let data = []
+    try {
+      data = await this.fetch(url)
+      console.log(data)
+    } catch (err) {
+      tracker.trackException(err.message)
+      bugsnag.notify(err)
+      this.setState({error: err.message})
+      console.warn(err)
+    }
+
+    this.setState(() => ({
+      loaded: true,
+      events: this.convertEvents(data),
+    }))
+  }
+
+  refresh = async () => {
+    let start = Date.now()
+    this.setState({refreshing: true})
+
+    await this.getEvents()
+
+    // wait 0.5 seconds – if we let it go at normal speed, it feels broken.
+    let elapsed = start - Date.now()
+    if (elapsed < 500) {
+      await delay(500 - elapsed)
+    }
+
+    this.setState({refreshing: false})
+  }
+
+  groupEvents = (events: EventType[]): any => {
+    const grouped = groupBy(events, event =>
+      event.pubDate.format('MMMM Do, YYYY'),
+    )
+
+    return toPairs(grouped).map(([title, data]) => ({title, data}))
+  }
+
+  onPressEvent = (event: ParsedPodcastEpisode) =>
+    this.props.navigation.navigate('ArchivedConvocationDetailView', {event})
+
+  renderSectionHeader = ({section: {title}}: any) =>
+    // the proper type is ({section: {title}}: {section: {title: string}})
+    <ListSectionHeader title={title} spacing={{left: 10}} />
+
+  renderItem = ({item}: {item: ParsedPodcastEpisode}) =>
+    <ArchivedConvocationRow onPress={this.onPressEvent} event={item} />
+
+  keyExtractor = (item: ParsedPodcastEpisode, index: number) => index.toString()
+
+  render() {
+    if (!this.state.loaded) {
+      return <LoadingView />
+    }
+
+    return (
+      <SectionList
+        ItemSeparatorComponent={ListSeparator}
+        ListEmptyComponent={<NoticeView text="No convocations." />}
+        style={styles.container}
+        sections={this.groupEvents(this.state.events)}
+        keyExtractor={this.keyExtractor}
+        refreshing={this.state.refreshing}
+        onRefresh={this.state.onRefresh}
+        renderSectionHeader={this.renderSectionHeader}
+        renderItem={this.renderItem}
+      />
+    )
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: c.white,
+  },
+  row: {
+    paddingTop: 5,
+    paddingBottom: 5,
+  },
+  '🎧': {
+    width: 200,
+    height: 100,
+  },
+  '📺': {
+    width: 200,
+    height: 100,
+  },
+})
+
+import Video from 'react-native-video'
+class ArchivedConvocationRow extends React.PureComponent {
+  props: {
+    event: ParsedPodcastEpisode,
+    onPress: ParsedPodcastEpisode => any,
+  }
+
+  _onPress = () => this.props.onPress(this.props.event)
+
+  render() {
+    const {event} = this.props
+
+    let annotation = event.enclosure.type.startsWith('audio/')
+      ? '🎧'
+      : event.enclosure.type.startsWith('video/') ? '📺' : ''
+
+    return (
+      <ListRow
+        contentContainerStyle={styles.row}
+        arrowPosition="center"
+        onPress={this._onPress}
+      >
+        <Title>{annotation} {event.title}</Title>
```

Comment:
I want to suggest that the annotation be part of the <ListSectionHeader> text since every convocation will either be a video or an audio recording and each will have its own section.